### PR TITLE
Cherry picked fixes from v0 to v1

### PR DESCRIPTION
**Pull request by abchatra**
These are ported fixes from v1. 

I haven't ported 3 commits which had lot of conflicts. But they are all doc fixes and I will let the owners port them one by one. 

Linting Csintro: 7bc3b6345c2e25c1ff5e4d7a29b5a0e9dc1ef0f8
Crashy bird: 5c9a70faff3481d6d3bbd3a56ba9f58b69d831b2
Change titles to title case in 'projects' (#1128): b193c2a1f10798f7fc935cba7d916447a694d50d

Part of change 
Cherry pick adding various cross-editor compat APIs (#863): 7308dbef23e9ee402bebb7e721d7014d8252c9e5

### CODE REVIEW

File: libs/core/pins.ts [hunk 1]
```diff
@@ -61,8 +61,137 @@ namespace pins {
...
+        packUnpackCore(format, nums, buf, true, offset)
+    }
+
+    export function unpackBuffer(format: string, buf: Buffer, offset = 0) {
+        let res: number[] = []
+        packUnpackCore(format, res, buf, false, offset)
+        return res
+    }
+
+    export class I2CDevice {
```
Comment (pelikhan) [inline libs/core/pins.ts]: already in i2c.ts

File: libs/core/pins.ts [hunk 2]
```diff
@@ -61,8 +61,137 @@ namespace pins {
...
+        let buf = createBuffer(packedSize(format))
+        packUnpackCore(format, nums, buf, true)
+        return buf
+    }
+
+    export function packIntoBuffer(format: string, buf: Buffer, offset: number, nums: number[]) {
+        packUnpackCore(format, nums, buf, true, offset)
+    }
+
+    export function unpackBuffer(format: string, buf: Buffer, offset = 0) {
```
Comment (pelikhan) [inline libs/core/pins.ts]: already in buffer.ts most likely

File: libs/core/pins.ts [hunk 3]
```diff
@@ -61,8 +61,137 @@ namespace pins {
         }
         return 0;
     }
+
```
Comment (pelikhan) [inline libs/core/pins.ts]: most of this stuff is in buffer.ts

**Comment by abchatra** [comment]
I reverted the two changes.

File: libs/core/pins.cpp [hunk 4]
```diff
@@ -381,9 +381,9 @@ namespace pins {
      * Write bytes to a 7-bit I2C `address`.
      */
     //%
-    int i2cWriteBuffer(int address, Buffer buf, bool repeat = false)
+    void i2cWriteBuffer(int address, Buffer buf, bool repeat = false)
     {
-      return uBit.i2c.write(address << 1, (char*)buf->payload, buf->length, repeat);
+      uBit.i2c.write(address << 1, (char*)buf->data, buf->length, repeat);
```
Comment (pelikhan) [inline libs/core/pins.cpp]: Changing the return value is a good change

**Comment by abchatra** [comment]
@pelikhan signoff I have reverted the changes which are not necessary and I have followup on those. I want to merge now and build a candidate for testing.

**Comment by abchatra** [comment]
Thanks @ganicke

**Comment by abchatra** [comment]
@pelikhan  https://github.com/Microsoft/pxt-microbit/pull/1476/commits/4e6ceb05a54f939f239822dfbe17b8c274e9fa14

**Pull request merged**